README.md

```diff
@@ -177,6 +177,181 @@ ApolloConverterFactory apolloConverterFactory = new ApolloConverterFactory.Build
+
+### Usage
+
+Raw HTTP Response Cache:
+```java
+//Directory where cached responses will be stored
+File file = new File("/cache/");
+
+//Size in bytes of the cache
+int size = 1024*1024;
+
+//Strategy for deciding when the cache becomes stale
+EvictionStrategy evictionStrategy = new TimeoutEvictionStrategy(5, TimeUnit.SECONDS);
+
+//Create the http response cache store
+ResponseCacheStore cacheStore = new DiskLruCacheStore(file, size); 
+
+//Build the Apollo Client
+ApolloClient apolloClient = ApolloClient.builder()
+                                    .serverUrl("/")
+                                    .httpCache(cacheStore, evictionStrategy)
+                                    .okHttpClient(okHttpClient)
+                                    .build();
+```
+
+Normalized Disk Cache:
+```java
+//Create the ApolloSqlHelper
+ApolloSqlHelper apolloSqlHelper = ApolloSqlHelper.create(context, "db_name");
+
+//Create NormalizedCacheFactory
+NormalizedCacheFactory cacheFactory = new SqlNormalizedCacheFactory(apolloSqlHelper);
+
+//Create the cache key resolver
+CacheKeyResolver<Map<String, Object>> resolver =  new CacheKeyResolver<Map<String, Object>>() {
+          @Nonnull @Override public CacheKey resolve(@Nonnull Map<String, Object> objectSource) {
+            String id = (String) objectSource.get("id");
+            if (id == null || id.isEmpty()) {
+              return CacheKey.NO_KEY;
+            }
+            return CacheKey.from(id);
+          }
+        }
+
+//Build the Apollo Client
+ApolloClient apolloClient = ApolloClient.builder()
+                                    .serverUrl("/")
+                                    .normalizedCache(cacheFactory, resolver)
+                                    .okHttpClient(okHttpClient)
+                                    .build();
+```
+
+Normalized InMemory Cache:
+```java
+
+//Create NormalizedCacheFactory
+NormalizedCacheFactory cacheFactory = new LruNormalizedCacheFactory(EvictionPolicy.builder().maxSizeBytes(10 * 1024).build());
+
+//Create the cache key resolver
+CacheKeyResolver<Map<String, Object>> resolver =  new CacheKeyResolver<Map<String, Object>>() {
+          @Nonnull @Override public CacheKey resolve(@Nonnull Map<String, Object> objectSource) {
+            String id = (String) objectSource.get("id");
+            if (id == null || id.isEmpty()) {
+              return CacheKey.NO_KEY;
+            }
+            return CacheKey.from(id);
+          }
+        }
+
+//Build the Apollo Client
+ApolloClient apolloClient = ApolloClient.builder()
+                                    .serverUrl("/")
+                                    .normalizedCache(cacheFactory, resolver)
+                                    .okHttpClient(okHttpClient)
+                                    .build();
+
+```
+
+For concrete examples of using response caches, please see the following classes in the [`apollo-integration`](apollo-integration) module:
```

Comment:
"For concrete examples of using response caches, please see the following tests"